src/main/java/com/onarandombox/MultiverseCore/configuration/DefaultMVConfig.java

```diff
@@ -0,0 +1,197 @@
+                                .build())
+                        .build())
+                .build();
+    }
+
+    private void migrateFromOldConfigFile() {
+        String content;
+        try {
+            content = Files.readString(configPath);
+        } catch (IOException e) {
+            return;
+        }
+        // Remove the old config section if it is still in the old ConfigurationSerializable.
+        if (content.contains("version: 2.5")) {
+            content = content.replace("==: com.onarandombox.MultiverseCore.MultiverseCoreConfiguration", "");
+        }
+        try {
+            Files.writeString(configPath, content);
+        } catch (IOException e) {
+            // ignore
+        }
+    }
+
+    public boolean load() {
+        return settings.load();
+    }
+
+    public void save() {
+        settings.save();
+    }
+
+    @Override
+    public void setEnforceAccess(boolean enforceAccess) {
+        settings.set(MVConfigNodes.ENFORCE_ACCESS, enforceAccess);
+    }
+
+    @Override
+    public boolean getEnforceAccess() {
+        return settings.get(MVConfigNodes.ENFORCE_ACCESS);
+    }
+
+    @Override
+    public void setAutoPurgeEnabled(boolean autopurge) {
+        settings.set(MVConfigNodes.AUTO_PURGE_ENTITIES, autopurge);
+    }
+
+    @Override
+    public boolean isAutoPurgeEnabled() {
+        return settings.get(MVConfigNodes.AUTO_PURGE_ENTITIES);
+    }
+
+    @Override
+    public void setTeleportIntercept(boolean teleportIntercept) {
+        settings.set(MVConfigNodes.TELEPORT_INTERCEPT, teleportIntercept);
+    }
+
+    @Override
+    public boolean getTeleportIntercept() {
+        return settings.get(MVConfigNodes.TELEPORT_INTERCEPT);
+    }
+
+    @Override
+    public void setFirstSpawnOverride(boolean firstSpawnOverride) {
+        settings.set(MVConfigNodes.FIRST_SPAWN_OVERRIDE, firstSpawnOverride);
+    }
+
+    @Override
+    public boolean getFirstSpawnOverride() {
+        return settings.get(MVConfigNodes.FIRST_SPAWN_OVERRIDE);
+    }
+
+    @Override
+    public void setFirstSpawnWorld(String firstSpawnWorld) {
+        settings.set(MVConfigNodes.FIRST_SPAWN_LOCATION, firstSpawnWorld);
+    }
+
+    @Override
+    public String getFirstSpawnWorld() {
+        return settings.get(MVConfigNodes.FIRST_SPAWN_LOCATION);
+    }
+
+    @Override
+    public void setUseCustomPortalSearch(boolean useDefaultPortalSearch) {
+        settings.set(MVConfigNodes.USE_CUSTOM_PORTAL_SEARCH, useDefaultPortalSearch);
+    }
+
+    @Override
+    public boolean isUsingCustomPortalSearch() {
+        return settings.get(MVConfigNodes.USE_CUSTOM_PORTAL_SEARCH);
+    }
+
+    @Override
+    public void setPortalSearchRadius(int searchRadius) {
+        settings.set(MVConfigNodes.CUSTOM_PORTAL_SEARCH_RADIUS, searchRadius);
+    }
+
+    @Override
+    public int getPortalSearchRadius() {
+        return settings.get(MVConfigNodes.CUSTOM_PORTAL_SEARCH_RADIUS);
+    }
```

Comment:
Crazy idea time.

I was reflecting on using more specific types to constrain allowable values. For example, having a type like PortalSearchRadius that can only represent valid values for it's use case. I guess that's maybe just 0 or higher for this case?

If we did use something like that, we could have a config system that could do something if the value isn't acceptable. For example, if this was set to -5, log a warning about this value and say that the default will be used instead. Then when we ask for this config value in the code, it's always going to be an valid value, no error handling needed. And if there is an error, it's likely that the valid value changed, in which case we can potentially only have to update the PortalSearchRadius type.